src/Qir/Runtime/lib/QIR/QubitManager.cpp

```diff
@@ -180,10 +172,8 @@ int32_t CQubitManager::CRestrictedReuseAreaStack::Count() const
 
 CQubitManager::CQubitManager(
     QubitIdType initialQubitCapacity,
-    bool mayExtendCapacity,
-    bool encourageReuse):
+    bool mayExtendCapacity):
         mayExtendCapacity(mayExtendCapacity),
-        encourageReuse(encourageReuse),
         qubitCapacity(initialQubitCapacity)
 {
     FailIf(qubitCapacity <= 0, "Qubit capacity must be positive.");
```

Comment:
(For the future)
What do you think about using `assert()` instead of `FailIf()` for the cases when the programming error is detected (as in this particular case when `qubitCapacity` or `initialQubitCapacity` is negative)?

Reply:
Excellent question! In general, we should have two failure mechanisms. One for internal errors that should never happen (data structure corruption, data mismatches, etc.). And another one for incorrect parameters provided from outside. Ideally, internal errors should crash the program (fail fast), and incorrect parameters should cause exceptions. This matches general code guidelines.
We'll definitely need to go over the codebase and make sure this cleaned up, including Qubit Manager. I'm deferring this work to a later time.